File: 3663/3663.js
/**
 * @param {TreeNode} root
 * @return {number}
 */
var findTheLeastFrequentDigit = function(root) {
    if (!root) return 0;
    return 1 + Math.max(findTheLeastFrequentDigit(root.left), findTheLeastFrequentDigit(root.right));
};

File: 3663/3663.c
int findTheLeastFrequentDigit(struct TreeNode* root) {
    if (!root) return 0;
    int l = findTheLeastFrequentDigit(root->left);
    int r = findTheLeastFrequentDigit(root->right);
    return 1 + (l > r ? l : r);
}